includes/class-wc-payments-vat-redirect-service.php

```diff
@@ -0,0 +1,63 @@
+<?php
+/**
+ * WooCommerce Payments VAT Redirect Service.
+ *
+ * @package WooCommerce\Payments
+ */
+
+namespace WCPay;
+
+defined( 'ABSPATH' ) || exit; // block direct access.
+
+/**
+ * Service class for handling VAT details redirect functionality.
+ *
+ * @todo This is a temporary solution that will be replaced by a dedicated VAT settings section.
+ *       Remove this class when the permanent solution is implemented.
+ */
+class WC_Payments_VAT_Redirect_Service {
+
+	/**
+	 * Initialize hooks.
+	 *
+	 * @return void
+	 */
+	public function init_hooks(): void {
+		add_action( 'template_redirect', [ $this, 'maybe_redirect' ], 1 );
+	}
+
+	/**
+	 * Handles redirection to VAT details modal if needed.
+	 *
+	 * @return void
+	 */
+	public function maybe_redirect(): void {
+		// phpcs:disable WordPress.Security.NonceVerification.Recommended
+		if ( empty( $_GET['vat-details-redirect'] ) ) {
```

Comment:
Nitpick: may give the param a prefix.

```suggestion
		if ( empty( $_GET['woopayments-vat-details-redirect'] ) ) {
```

Reply:
Done here https://github.com/Automattic/woocommerce-payments/pull/10859/commits/5ba8610c2bb8fb0bc5e0d118db721c60e253c5ea